cardano-launcher/src/Cardano/Shell/CLI.hs

```diff
@@ -0,0 +1,60 @@
+module Cardano.Shell.CLI
+    ( getLauncherOptions
+    ) where
+
+import           Cardano.Prelude
+
+import           Data.Yaml (ParseException, decodeFileEither)
+import           Options.Applicative (Parser, ParserInfo, execParser, fullDesc,
+                                      header, help, helper, info, long, metavar,
+                                      progDesc, short, strOption, value)
+import           System.Environment (getExecutablePath)
+import           System.FilePath (takeDirectory, (</>))
+
+import           Cardano.Shell.Launcher
+
+-- | Path to launcher-config.yaml file
+newtype LauncherOptionPath = LauncherOptionPath
+    { getLauncherOptionPath  :: FilePath
+    }
+
+-- | Default path to the launcher-config.yaml file
+--
+-- This will be used if the filepath was not provided via CLI
+getDefaultConfigPath :: IO FilePath
+getDefaultConfigPath = do
+    launcherDir <- takeDirectory <$> getExecutablePath
```

Comment:
Are you sure about this one? Did you test this somehow? I presume that the executable will be provided once we create an executable out of it, but I'm still not sure which one.
```
$> stack ghci

> import           System.Environment (getExecutablePath)
> getExecutablePath
"/home/ksaric/.stack/programs/x86_64-linux/ghc-8.6.5/lib/ghc-8.6.5/bin/ghc"
```

Reply:
This is used for fetching configuration file on both Mac and Windows.

Reply:
I understand that, what I'm asking is what path will be returned here if we run it using Nix, for example, and would that work?

Reply:
that is why linux uses the `--config` argument on the command line, rather then expecting the cfg to be in the same dir as the binary

https://github.com/input-output-hk/daedalus/blob/develop/installers/nix/linux.nix#L47-L48